mixer/pkg/runtime/routing/builder.go

```diff
@@ -575,9 +575,11 @@ func (b *builder) getBuilderAndMapperDynamic(
 	var instBuilder template.InstanceBuilderFn = func(attrs attribute.Bag) (interface{}, error) {
 		var err error
 		ba := make([]byte, 0, defaultInstanceSize)
-		// The encoder produces
-		if ba, err = instance.Encoder.Encode(attrs, ba); err != nil {
-			return nil, err
+		if instance.Encoder != nil {
```

Comment:
Should we produce an error here instead?

Reply:
I think it's fine. Encoder is nil when template is empty, such as checknothing. The logic here is to bear this corner case. Alternative is to make encoder always initialized as a noop one when template is empty, which requires some code change. So I'd rather keep it simple as this.